synapse/lib/cell.py

```diff
@@ -666,6 +673,17 @@ def addHttpApi(self, path, ctor, info):
     def getLocalUrl(self, share='*', user='root'):
         return f'cell://{user}@{self.dirn}:{share}'
 
+    def _initCellConf(self, conf: c_abc.MutableMapping):
+        if isinstance(conf, dict):
+            conf = s_config.Config020(schema=self.confschema,
+                                      conf=conf,
+                                      )
+        yaml_conf = self._loadCellYaml('cell.yaml')
+        for k, v in yaml_conf.items():
```

Comment:
`yaml_conf.update(conf)`

Reply:
Using update as opposed to setdefault is that current values are overridden.  From previous discussion, the precendence order of configuration values was to be:

1. command line
2. envars
3. cell.yaml (configuration on disk).

items 1 & 2 effectively have to come in outside of the cell via the conf argument; and inside of the cell we only pull in data from the .yaml file.  By using setdefault, we can control which order values come into the configuration diction, whereas update will smash over existing values.